### PR TITLE
Fix is_learner attribute and admin login

### LINKED ISSUE #7199: SignInPage does not implement the "optional learner password" facility setting
### Observed behavior

Screenshot 1: 

I try to login as a learner `l`, but am brought to this second "Create a password" page. (Also see other issues pointed out in screenshot)

image

Screenshot 2:

I try to login as superuser `jb`. I get stuck at the sign-in page

image

### User-facing consequences

…

### Errors and logs

…

### Steps to reproduce

…

### Context
0.14 beta 6

### PR DESCRIPTION
**Pull request by jredrejo**
### Summary

Fixes the two issues described at #7199, all happening when the facility setup allowing users to login without password is set:

- Self-created users are requested to add a password when trying to login
- admin login becomes impossible

### Reviewer guidance
In the facility settings, allow the users to create their own accounts and to login without password.
1. Try to login as an admin
2. Create a new user and try to login using it without password

### References
Closes #7199 , #7155 

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/auth/api.py [hunk 1]
```diff
@@ -218,7 +218,7 @@ def users_for_facilities(self, request):
 
         users = (
             FacilityUser.objects.filter(facility__in=facility_ids)
-            .annotate(is_learner=Exists(Role.objects.filter(user=OuterRef("id"))))
+            .annotate(is_learner=~Exists(Role.objects.filter(user=OuterRef("id"))))
```
Comment (rtibbles) [inline kolibri/core/auth/api.py]: That's an important character!

File: kolibri/plugins/user/assets/src/views/SignInPage.vue [hunk 2]
```diff
@@ -282,11 +282,12 @@
       ...mapState(['facilityId']), // backend's default facility on load
       ...mapState('signIn', ['hasMultipleFacilities', 'usersForSelectedFacilities']),
       ...mapState({
+        passwordMissing: state => state.core.loginError === LoginErrors.PASSWORD_MISSING,
         invalidCredentials: state => state.core.loginError === LoginErrors.INVALID_CREDENTIALS,
         busy: state => state.core.signInBusy,
       }),
       simpleSignIn() {
-        return this.facilityConfig.learner_can_login_with_no_password;
+        return this.facilityConfig.learner_can_login_with_no_password && !this.passwordMissing;
```
Comment (rtibbles) [inline kolibri/plugins/user/assets/src/views/SignInPage.vue]: To clarify here - this is being toggled off in the case that a non-learner has tried to login without the password, and then the backend has reported that the password is needed?

**Review by rtibbles — COMMENTED** [review]
Just one question, but it seems to make sense to me, have not manually tested.

File: kolibri/plugins/user/assets/src/views/SignInPage.vue (hunk 2 above)
Comment (jredrejo) [reply kolibri/plugins/user/assets/src/views/SignInPage.vue]: I have moved it to the computed property to decide if the password must be shown, to be more semantic (if that can be said in English)

Comment (rtibbles) [reply kolibri/plugins/user/assets/src/views/SignInPage.vue]: Yep - perfectly semantic use of semantic for me :)

**Comment by rtibbles** [comment]
Apologies, @jredrejo - I was testing this locally, and things were behaving very much not as I expected, so I built on top of your fixes.

**Comment by nucleogenesis** [comment]
The response speed of clicking a button for a user who needs to make a password is very short (< 1s) but also very noticeable - to the point that on slower hardware it could be > 1s. If I put "Slow 3G" in dev tools it takes ~1-2s.

The wait isn't a problem - but it might be worth having a loading transition.

slow-response-when-selecting-user

**Comment by nucleogenesis** [comment]
@rtibbles It may be worth addressing the issues this PR is targeting during @jredrejo and my overhaul based on the new designs from Riddhi.

I've encountered a few weird issues - all of which seem to stem from how the "create a password" page is handled.

IF you click a username on the User List Flow, 
AND that user has `NOT_SPECIFIED` a password
AND the setting is that you must have a password
THEN the learner is taken to the "create a new password" page
BUT THEN the learner is logged in regardless of what happens on the "create a new password" page. Refresh the page - you're logged in. Go back to "select a user" and refresh - you're logged in as whatever user you selected in the first place.

This shouldn't be possible because they are required to have a password.

Additionally, if you've gotten into that "Create a password" page, the rest of the flow breaks. Go back and change your facility and it doesn't change your facility - keeps you on the current facility ("Sign in to 'Facility 1'" - even when you selected to go to Facility 2) - but lists no users when the user list is supposed to show. If you refresh - you're logged in as whatever user you selected to get you to "Create a password" in the first place - but once you log out you're in the facility you selected before it broke (meaning the value is at least being saved in the local storage).

Here is a gif showing some of what I'm talking about.

screencast-localhost_8000-2020 07 01-14_55_39

**Review by nucleogenesis — CHANGES_REQUESTED** [review]
Left some comments and illustrative gifs

**Comment by rtibbles** [comment]
Issues highlighted in review should be addressed!

**Review by nucleogenesis — APPROVED** [review]
LGTM

Thanks @rtibbles - @jredrejo and I should be smooth sailing from here.

**Pull request merged**